cli/klaus/src/main.rs

```diff
@@ -76,44 +76,50 @@ async fn main() -> Result<(), Error> {
     let span = tracing::Span::current();
     span.record("pid", args.pid);
     span.record("registry", traceable_path(&args.registry_path));
-    span.record("socket", maybe_traceable_path(&args.socket_path));
+    span.record("dot_flox_hash", &args.dot_flox_hash);
+    span.record("socket", traceable_path(&args.socket_path));
     span.record("log", maybe_traceable_path(&args.log_path));
-    if let Some(ref path) = args.socket_path {
-        debug!(socket_path = traceable_path(&path), "was provided a socket");
-    }
 
     debug!("starting");
+    debug!(
+        path = traceable_path(&args.socket_path),
+        exists = &args.socket_path.exists(),
+        "checked socket"
+    );
 
-    // The parent may have already died, in which case we just want to exit
-    if let Some(ref pid) = args.pid {
-        // TODO: re-use the method from ActivationPid after merging both PRs
-        let parent_is_running = match kill(Pid::from_raw(*pid), None) {
-            // These semantics come from kill(2).
-            Ok(_) => true,              // Process received the signal and is running.
-            Err(Errno::EPERM) => true,  // No permission to send a signal but we know it's running.
-            Err(Errno::ESRCH) => false, // No process running to receive the signal.
-            Err(_) => false,            // Unknown error, assume no running process.
-        };
-        if !parent_is_running {
-            return Err(anyhow!("detected that watchdog had unexpected parent"));
-        }
+    // Register activation PID so that we can track last one out
+    let activation = ActivationPid::from(args.pid);
+    if !activation.is_current_process_parent() {
+        return Err(anyhow!("detected that watchdog had unexpected parent"));
```

Comment:
I've added a note to https://github.com/flox/flox/issues/1588 for whether we do cleanup in this case.